### ISSUE TITLE
How to generate new sample sentence?

### ISSUE THREAD

**Issue by ydwang72** [CSTR-Edinburgh/merlin#28]
using model trained by 'slt_arctic_full_data'?

**Comment by ronanki** [comment]
1. The scripts are under development. However, you can use festival front-end to generate the full-contextual labels similar to the current format. 

2. Then, please follow the step 4 in https://github.com/CSTR-Edinburgh/merlin/blob/master/egs/slt_arctic/s1/run_full_voice.sh

**Comment by ronanki** [comment]
The scripts are ready now -- to generate new test sentences. 

Steps to follow:
-------------------
Step 1: prepare a text folder, with each file containing a new test sentence. 
Step 2: run below script
https://github.com/CSTR-Edinburgh/merlin/blob/master/egs/slt_arctic/s1/scripts/prepare_labels_from_txt.sh
Step 3: Then, please follow the step 4 in below script:
https://github.com/CSTR-Edinburgh/merlin/blob/master/egs/slt_arctic/s1/run_full_voice.sh 

Hope these steps are useful to create novel sentences.

**Comment by ydwang72** [comment]
Srikanth,

Thank you. It's working as you guided!

Just to clarify Step 1 for others, the place of your new 'txt' folder should be at:
experiments/slt_arctic_full/test_synthesis/txt

**Comment by uvaidya** [comment]
Hi Srikanth, ydwang72,
I wanted to use trained model obtained by running run_demo.sh/run_full_voice.sh for the purpose of synthesizing audio for new sentences.
Towards doing above, i followed the steps you mentioned in "Steps to follow above" but while running merlin_synthesis.sh, i get following error :-  
_"2017-04-06 05:54:15,554 DEBUG   param_generation: acoustic_decomposition for 1 files
2017-04-06 05:54:15,554 INFO    param_generation: processing    1 of    1: /media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/egs/slt_arctic/s1/experiments/slt_arctic_demo/test_synthesis/wav/own_01.cmp
2017-04-06 05:54:15,554 DEBUG   param_generation:  feature: mgc
checking shape of features over here in generation ...  (0, 180)
2017-04-06 05:54:15,554 DEBUG   param_generation: starting MLParameterGeneration.generation
************************************checking shape of features over here in generation ...*************************************************** 
Traceback (most recent call last):
  File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/run_merlin.py", line 1175, in <module>
    main_function(cfg)
  File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/run_merlin.py", line 943, in main_function
    generator.acoustic_decomposition(gen_file_list, cfg.cmp_dim, cfg.out_dimension_dict, cfg.file_extension_dict, var_file_dict, do_MLPG=cfg.do_MLPG, cfg=cfg)    
  File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/frontend/parameter_generation.py", line 161, in acoustic_decomposition
    gen_features = mlpg_algo.generation(current_features, var, out_dimension_dict[feature_name]/3)
  File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/frontend/mlpg_fast.py", line 122, in generation
    var_frames[0, 1] = 100000000000;
**IndexError: index 0 is out of bounds for axis 0 with size 0**
Lock freed_
"
I am attaching logs generated while running merlin_synthesis.py
[command_line_logs.txt](https://github.com/CSTR-Edinburgh/merlin/files/901133/command_line_logs.txt)

I created a directory by the name of txt in path mentioned by ydwang above, and created a test file by the name of own_01.txt which has "Summer vacations has started." as a text in it.

Please help me with this issue, by letting me know what i am missing.

Thanks
Utkarsh Vaidya

**Comment by fizerkhan** [comment]
When I run the first command "https://github.com/CSTR-Edinburgh/merlin/blob/master/egs/slt_arctic/s1/scripts/prepare_labels_from_txt.sh"
I got the following error 

```
Please configure festival path in scripts/prepare_labels_from_txt.sh !!
```

How do install install/setup festival?

**Comment by ronanki** [comment]
sudo apt-get install festival festvox-kallpc16k

**Comment by fizerkhan** [comment]
@ronanki Great. Thanks.

Do you have any setup instructions to run this? I spent two days to figure out how to run this, but I could not make it.

It will be very helpful if someone put instructions to run this.

**Comment by ronanki** [comment]
If you install festival -- it should be straight-forward. All you need to do is run this [script](https://github.com/CSTR-Edinburgh/merlin/blob/master/egs/slt_arctic/s1/merlin_synthesis.sh).

**Comment by uvaidya** [comment]
Hi @fizerkhan , 
Sorry for late reply
#96 will help you with mostly all required instructions need to be followed.

Regards

**Comment by fizerkhan** [comment]
@uvaidya Thanks a lot for the update 👍 

It is very hard to follow the instructions from that issue. Can you please add those instructions into separate file e.g SETUP.md or README.md?

**P.S:** The instructions are not only for festival dependencies. The instructions must guide the setup from the scratch.

**Comment by b-schnell** [comment]
Hey, I have a text file with the only sentence
_Will we ever forget it._
in the txt folder. But when using the merlin_synthesis.sh script the first world is skipped. Is the first word reserved for the utterance id? When I'm using
_arctic_a0005 Will we ever forget it._
only _arc_ is skipped.

I found the same behaviour in the example wavs _arctic_b0537_ and _arctic_b0538_. However, the examples _arctic_b0535, arctic_b0536 and arctic_b0359_ seem to be correct.

 The same problem occurs with the utts.data file with the line
_( arctic_a0005 "Will we ever forget it." )_

Am I using a wrong format?

**Comment by ronanki** [comment]
The format seems right and is similar to [cmudata](http://festvox.org/cmu_arctic/cmuarctic.data). If you are not sure of the format, please use simple plain text files in a directory as input.

**Comment by b-schnell** [comment]
In the attached file ([arctic_a0005.txt](https://github.com/CSTR-Edinburgh/merlin/files/1124969/arctic_a0005.txt)) I'm using
_will we ever forget it_
The audio is created with ./merlin_synthesis.sh, but the first world is missing. Can you reproduce that?

**Comment by xiaobing88** [comment]
I have slt_arctic_demo.
now I want to my text to speech.
There have utts.data('( arctic_a0001 "This is what I can say" ) ') and text(a dir of have one text 1.txt(i am a boy)) in test_synthesis.
but i meet a  problem.
**[root@localhost s1]# ./merlin_synthesis.sh 
Step 1: creating label files from text...
Please configure festival path in scripts/prepare_labels_from_txt.sh !!
Step 1 not successful !!
**
**[root@localhost scripts]# ./prepare_labels_from_txt.sh 
Usage: ./scripts/prepare_labels_from_txt.sh conf/global_settings.cfg
[root@localhost scripts]# cd ../
[root@localhost s1]# ./scripts/prepare_labels_from_txt.sh conf/global_settings.cfg 
Please configure festival path in scripts/prepare_labels_from_txt.sh !!
**
how to configure festival path?
I really don't know. 
i hope you can help me 
Thanks.

**Comment by b-schnell** [comment]
Have a look at #96

**Comment by mrgloom** [comment]
Under Ubuntu 16 after running `sudo apt-get install festival festvox-kallpc16k` I still get 

```
Step 1: creating label files from text...
Please configure festival path in scripts/prepare_labels_from_txt.sh !!
Step 1 not successful !!
```

when i'm trying to run `merlin/egs/slt_arctic/s1$ ./merlin_synthesis.sh`

**Comment by Sumaid** [comment]
> When i'm trying to run `merlin/egs/slt_arctic/s1$ ./merlin_synthesis.sh`
> under Ubuntu 16 after running `sudo apt-get install festival festvox-kallpc16k` I still get
> 
> ```
> Step 1: creating label files from text...
> Please configure festival path in scripts/prepare_labels_from_txt.sh !!
> Step 1 not successful !!
> ```
> 
> I have added 1 .txt file for test `echo "Hello, I love you!" > experiments/slt_arctic_demo/test_synthesis/txt/test_me.txt`
> 
> Festival is visible in system
> 
> ```
> which festival
> /usr/bin/festival
> ```
> 
> **UDPATE:**
> 
> Solution was simple just `bash tools/compile_other_speech_tools.sh`



> Hi Srikanth, ydwang72,
> I wanted to use trained model obtained by running run_demo.sh/run_full_voice.sh for the purpose of synthesizing audio for new sentences.
> Towards doing above, i followed the steps you mentioned in "Steps to follow above" but while running merlin_synthesis.sh, i get following error :-
> _"2017-04-06 05:54:15,554 DEBUG param_generation: acoustic_decomposition for 1 files 2017-04-06 05:54:15,554 INFO param_generation: processing 1 of 1: /media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/egs/slt_arctic/s1/experiments/slt_arctic_demo/test_synthesis/wav/own_01.cmp 2017-04-06 05:54:15,554 DEBUG param_generation: feature: mgc checking shape of features over here in generation ... (0, 180) 2017-04-06 05:54:15,554 DEBUG param_generation: starting MLParameterGeneration.generation ************************************checking shape of features over here in generation ...*************************************************** Traceback (most recent call last): File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/run_merlin.py", line 1175, in main_function(cfg) File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/run_merlin.py", line 943, in main_function generator.acoustic_decomposition(gen_file_list, cfg.cmp_dim, cfg.out_dimension_dict, cfg.file_extension_dict, var_file_dict, do_MLPG=cfg.do_MLPG, cfg=cfg) File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/frontend/parameter_generation.py", line 161, in acoustic_decomposition gen_features = mlpg_algo.generation(current_features, var, out_dimension_dict[feature_name]/3) File "/media/dl_audio/hdd2/Utkarsh/tts_exp/merlin/src/frontend/mlpg_fast.py", line 122, in generation var_frames[0, 1] = 100000000000; **IndexError: index 0 is out of bounds for axis 0 with size 0** Lock freed_
> "
> I am attaching logs generated while running merlin_synthesis.py
> [command_line_logs.txt](https://github.com/CSTR-Edinburgh/merlin/files/901133/command_line_logs.txt)
> 
> I created a directory by the name of txt in path mentioned by ydwang above, and created a test file by the name of own_01.txt which has "Summer vacations has started." as a text in it.
> 
> Please help me with this issue, by letting me know what i am missing.
> 
> Thanks
> Utkarsh Vaidya

@uvaidya I am getting the same error. Did you figure out how to fix it?

@ronanki Can you please check what to do?
